### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by tamird**
Extracted from #4536.

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/4595)

### CODE REVIEW

File: build/circle-deps.sh [hunk 1]
```diff
@@ -36,6 +36,9 @@ function fetch_docker() {
 # the argument causing the commands in the if-branch to be executed
 # within the docker container.
 if [ "${1-}" = "docker" ]; then
+  go get -u github.com/robfig/glock
+  glock sync -n < GLOCKFILE
```
Comment (tschottdorf) [inline build/circle-deps.sh]: was there a particular reason for skipping the commands? Do they take up time on every invocation now?
Hmm. looks like they're cached below.

**Comment by tschottdorf** [comment]
LGTM but not sure what the original reason for grepping out cmds was.

**Comment by tamird** [comment]
Yeah, I'm not sure either.

**Comment by petermattis** [comment]
Review status: 0 of 4 files reviewed at latest revision, 1 unresolved discussion.

---

*[build/circle-deps.sh, line 40 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4595#-KBDjy_suJPeAB_5MKr6-r1-40:-KBDldXvDqj4U0OoLM9z:924170424) ([raw file](https://github.com/cockroachdb/cockroach/blob/711d877e453cd97864d4f12644bdea773af7bdff/build/circle-deps.sh#L40)):*
You're install the commands on every shard now instead of only on `shard 2` where the commands are used. Was the previous code actually causing problems?

There was a good reason at one point for doing the `glock sync` outside of docker. Do you know why this is possible now? 

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4595)*

**Comment by tamird** [comment]
Review status: 0 of 4 files reviewed at latest revision, 1 unresolved discussion.

---

*[build/circle-deps.sh, line 40 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4595#-KBDjy_suJPeAB_5MKr6-r1-40:-KBDnAr0bPXpekE2Bqss:-1993139188) ([raw file](https://github.com/cockroachdb/cockroach/blob/711d877e453cd97864d4f12644bdea773af7bdff/build/circle-deps.sh#L40)):*
The previous code had a rare case of a git error during `glock sync` when building without cache. I wanted to move the `glock sync` into the container so that we can control the git version used. I have no idea why `glock sync` was previously being done outside the container, or what has changed since then.

You're right that the commands are being installed on every shard now, but aren't they all cached? The potential savings don't seem to be worth the complexity.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4595)*

**Comment by petermattis** [comment]
LGTM

---

Review status: 0 of 4 files reviewed at latest revision, 1 unresolved discussion.

---

*[build/circle-deps.sh, line 40 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4595#-KBDjy_suJPeAB_5MKr6-r1-40:-KBDo0ODN_X3a1jDVKAN:-494955710) ([raw file](https://github.com/cockroachdb/cockroach/blob/711d877e453cd97864d4f12644bdea773af7bdff/build/circle-deps.sh#L40)):*
Mucking with a `circle-deps.sh` is a good way to waste an afternoon fixing problems after your initial change. Just saying. I don't see a problem with this change, but that might just be a failure of my imagination. 

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4595#-:-KBDof-IjelZsKfOEKuS:676834233)*

**Comment by tamird** [comment]
Review status: 0 of 4 files reviewed at latest revision, 1 unresolved discussion.

---

*[build/circle-deps.sh, line 40 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4595#-KBDjy_suJPeAB_5MKr6-r1-40:-KBDpHgnSi-eF_xqW2sv:-1058568776) ([raw file](https://github.com/cockroachdb/cockroach/blob/711d877e453cd97864d4f12644bdea773af7bdff/build/circle-deps.sh#L40)):*
Agreed - I only mucked with this after I saw weirdness in the build in #4536, and I've had about a dozen builds since then. Fingers crossed.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4595)*

**Pull request merged**